pkg/pool-stable-phantom/test/StablePhantomPool.test.ts

```diff
@@ -641,17 +640,168 @@ describe('StablePhantomPool', () => {
+      });
+
+      it('fails if wrong user data', async () => {
+        const wrongUserData = ethers.utils.defaultAbiCoder.encode(['address'], [lp.address]);
+
+        await expect(pool.exit({ data: wrongUserData })).to.be.revertedWith('Transaction reverted without a reason');
+      });
+
+      describe('exit exact BPT in for all tokens out', () => {
+        context('not in recovery mode', () => {
+          itExitsExactBptInForAllTokensOutProperly();
         });
+
+        context('in recovery mode', () => {
+          sharedBeforeEach('enable recovery mode', async () => {
+            await pool.enableRecoveryMode(admin);
+          });
+
+          itExitsExactBptInForAllTokensOutProperly();
+        });
+
+        function itExitsExactBptInForAllTokensOutProperly() {
+          it('grants all tokens for exact bpt', async () => {
+            // Exit with half of the BPT balance
+            const bptIn = previousBptBalance.div(2);
+            const expectedAmountsOut = initialBalances.map((balance) => bn(balance).div(2));
+
+            const result = await pool.multiExitGivenIn({ from: lp, bptIn });
+
+            // Balances are reduced by half because we are returning half of the BPT supply
+            expect(result.amountsOut).to.be.equalWithError(expectedAmountsOut, 0.001);
+
+            // Current BPT balance should have been reduced by half
+            expect(await pool.balanceOf(lp)).to.be.equalWithError(bptIn, 0.001);
+          });
+
+          it('fully exit', async () => {
+            // The LP doesn't own all BPT, since some was locked. They will only be able to extract a (large) percentage
+            // of the Pool's balance: the rest remains there forever.
+            const totalBPT = await pool.getVirtualSupply();
+            const expectedAmountsOut = initialBalances.map((balance) =>
+              bn(balance).mul(previousBptBalance).div(totalBPT)
+            );
+
+            const result = await pool.multiExitGivenIn({ from: lp, bptIn: previousBptBalance });
+
+            // All balances are extracted
+            expect(result.amountsOut).to.be.lteWithError(expectedAmountsOut, 0.00001);
+
+            // Current BPT balances should be zero due to full exit
+            expect(await pool.balanceOf(lp)).to.equal(0);
+          });
+
+          async function multiExitGivenIn(): Promise<void> {
+            const totalBPT = await pool.getVirtualSupply();
+            const expectedAmountsOut = initialBalances.map((balance) =>
+              bn(balance).mul(previousBptBalance).div(totalBPT)
+            );
+
+            const result = await pool.queryMultiExitGivenIn({ bptIn: previousBptBalance });
+
+            expect(result.bptIn).to.equal(previousBptBalance);
+            expect(result.amountsOut).to.be.lteWithError(expectedAmountsOut, 0.00001);
+          }
+
+          it('can tell how many tokens it will give in return', async () => {
+            await multiExitGivenIn();
+          });
```

Comment:
```suggestion
          it('can tell how many tokens it will give in return', async () => {
                        const totalBPT = await pool.getVirtualSupply();
            const expectedAmountsOut = initialBalances.map((balance) =>
              bn(balance).mul(previousBptBalance).div(totalBPT)
            );

            const result = await pool.queryMultiExitGivenIn({ bptIn: previousBptBalance });

            expect(result.bptIn).to.equal(previousBptBalance);
            expect(result.amountsOut).to.be.lteWithError(expectedAmountsOut, 0.00001);
          });
```

Reply:
Same as in the other one :)